core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/sortPartitions/SortPartitionOpExec.scala

```diff
@@ -0,0 +1,198 @@
+    val skewedFutures = new ArrayBuffer[Future[Boolean]]()
+    val listsToSend = getListsToSendToSkewed()
+    listsToSend.foreach(list => {
+      skewedFutures.append(
+        asyncRPCClient.send(
+          AcceptMutableState(list, listsToSend.size),
+          skewedWorkerIdentity
+        )
+      )
+    })
+    Future
+      .collect(skewedFutures)
+      .onSuccess(s => {
+        if (s.contains(false)) {
+          throw new WorkflowRuntimeException("Error in sending tuples to skewed worker of Sort")
+        }
+      })
+  }
+
+  def sortTuples(): Iterator[Tuple] = ownTuples.sortWith(compareTuples).iterator
+
+  override def processTexeraTuple(
+      tuple: Either[Tuple, InputExhausted],
+      input: LinkIdentity,
+      pauseManager: PauseManager,
+      asyncRPCClient: AsyncRPCClient
+  ): Iterator[Tuple] = {
+    tuple match {
+      case Left(t) =>
+        if (Constants.reshapeSkewHandlingEnabled) {
+          val attributeType = t.getSchema().getAttribute(sortAttributeName).getType()
+          val attributeIndex = t.getSchema().getIndex(sortAttributeName)
+          var workerIdxForKey = -1
+          attributeType match {
+            case AttributeType.LONG =>
+              workerIdxForKey = getWorkerIdxForKey(t.getLong(attributeIndex))
+            case AttributeType.INTEGER =>
+              workerIdxForKey = getWorkerIdxForKey(t.getInt(attributeIndex).toLong)
+            case AttributeType.DOUBLE =>
+              workerIdxForKey = getWorkerIdxForKey(t.getDouble(attributeIndex).toLong)
+            case _ =>
+              throw new RuntimeException(
+                "unsupported attribute type in SortOpExec: " + attributeType.toString()
+              )
+          }
+          if (workerIdxForKey == localIdx) {
+            ownTuples.append(t)
+          } else {
+            skewedWorkerTuples.append(t)
+          }
+        } else {
+          ownTuples.append(t)
+        }
+
+        Iterator()
+      case Right(_) =>
+        if (Constants.reshapeSkewHandlingEnabled) {
+          if (helperWorkerIdentity == null && skewedWorkerIdentity == null) {
+            // this worker is neither the skewed worker nor the helper worker
+            sortTuples()
+          } else if (helperWorkerIdentity != null) {
+            // this worker is the skewed worker.
+            if (!waitingForTuplesFromHelper) {
+              // It has received the state and can output the results
+              sortTuples()
+            } else {
+              // It will pause its execution here. The execution will be resumed once the state is received
+              // from the helper worker
+              pauseManager.pausedByOperatorLogic = true
+              Iterator()
+            }
+          } else if (skewedWorkerIdentity != null) {
+            // this worker is the helper worker. It needs to send state to the skewed worker.
+            sendTuplesToSkewedWorker(asyncRPCClient)
+            sortTuples()
```

Comment:
why call sortTuples here?

Reply:
because helper worker should output its own tuples.